File: Programmers/연습문제/Level 1/문자열 내 p와 y의 개수/Solution.js
function solution(s){
    const lower_s = getLowerCase(s);
    const counter = getCounter(lower_s);
    const answer = getAnswer(counter);
    return answer;
}

function getLowerCase(str) {
    return str.toLowerCase();
}

function getCounter(str) {
    let counter = new Map();
    for (const char of Array.from(str)) {
        counter.set(char, counter.get(char) + 1 || 1);
    }
    return counter;
}

function getAnswer(counter) {
    return counter.get('p') === counter.get('y');
}

File: Programmers/연습문제/Level 1/문자열 내 p와 y의 개수/Solution.swift
func solution(_ s:String) -> Bool {
    var counter: [Character: Int] = [:]
    for char in s {
        counter[char, default: 0] += 1
    }
    return getValue(counter, "p") + getValue(counter, "P") == getValue(counter, "y") + getValue(counter, "Y")
}

func getValue(_ counter: Dictionary<Character, Int>, _ key: Character) -> Int {
    return counter[key] != nil ? counter[key]! : 0
}


// TC
let s = "oooyY"
print(solution(s))
